Integrate CLI autocompletion into debian and RPM packages

hairyhum opened the pull request:
Fixes #12.

Installs autocompletion script into `/et/profile.d/`. The script will detect a shell type and install completions accordingly.

Commit:
Install auto-complete script

hairyhum committed:
Fix autocomplete scripts file names

hairyhum committed:
RPM paths for autocomplete scripts

Commit:
Delete autocomplete script when removing deb package

Commit:
Make sure completion call has locale var

michaelklishin committed:
Merge branch 'master' into rabbitmq-server-release-12

michaelklishin reviewed (changes requested):
I couldn't get this to work with either bash or zsh, as a regular user or `root`. Completion list never pops up.

I also had to fix `--auto-complete` in the CLI repo because that didn't match and work as expected: https://github.com/rabbitmq/rabbitmq-cli/commit/283b18e65aab35426e2957b5602f6519ace18233.

Commit:
Update profile after installation to enable autocompletion

Commit:
Pass auto-completion arguments as argv, not a single argument

Commit:
Source /etc/profile with shell builtin

Commit:
Restore accidentally deleted gitignore

Commit:
Install separate autocomplete scripts for BASH and ZSH

    . /etc/profile can be called from shells incompatible with
    autocomplete script syntax. To avoid that specific BASH and
    ZSH autocomplete scripts are called only if shell version is
    detected.

Commit:
Fix autocomplete script paths

Commit:
Reworked ZSH completion. It's now a function that should explicitely loaded on startup

Commit:
Print a hint to enable zsh autocompletion

hairyhum commented:
Apparently ZSH completion is not that easy.
ZSH completion script is now a function, that should be loaded explicitly by `autoload _enable_rabbitmqctl_completion; _enable_rabbitmqctl_completion`

michaelklishin committed:
Merge branch 'master' into rabbitmq-server-release-12

Commit:
Instruction to complete rabbtimq-plugins and rabbitmq-diagnostics

michaelklishin reviewed (approved):
I uncovered a curious exception with zsh but it's not strictly autocompletion related.
